pkg/processor/runtime/rpc/connection/abstract.go

```diff
@@ -538,6 +698,12 @@ func (bc *AbstractControlMessageConnection) SetBroker(abstractBroker *controlcom
 		abstractBroker)
 }
 
+func (bc *AbstractControlMessageConnection) Stop() {
+	bc.stopOnce.Do(func() {
```

Comment:
why is `Once` needed here too?

Reply:
@TomerShor to avoid stopping it more than once (if we stop from watcher and termination flow at the same time)